api/v1beta2/tenant_types.go

```diff
@@ -56,6 +56,12 @@ type TenantSpec struct {
 	// When enabled, the deletion request will be declined.
 	//+kubebuilder:default:=false
 	PreventDeletion bool `json:"preventDeletion,omitempty"`
+	// Use this if you want to enforce the Tenant name prefix to specific Tenants instead of globally with forceTenantPrefix in CapsuleConfiguration.
+	// Enforces Namespaces created for this Tenant to be named with the Tenant name prefix,
+	// separated by a dash (i.e. for Tenant 'foo', namespace names must be prefixed with 'foo-').
+	// This is useful to avoid Namespace name collision.
+	// +kubebuilder:default=false
```

Comment:
How about:
```yaml
// With this feature you can opt-in at the Tenant level, if the global configuration for forceTenantPrefix is false.
```

Since you can't opt-out, global capsule forceTenantPrefix will override if set to true (setting to false has no effect in tenant in that case)

Reply:
@oliverbaehler @prometherion what are your thoughts?

Reply:
I'm a bit worried this new feature could raise more confusion about the feature itself.

I'm wondering if it's better to have this as an annotation, mostly because who really needs it would require digging into the documentation, and purposely setting it, rather than reading the specification and misunderstanding it.

Reply:
Annotation can make sense to avoid confusion.
But, more to my question on the behaviour - do we want this setting to override global prefix for a Tenant - allowing to disable even if globally enabled OR just allowing to enable for a Tenant if globally disabled (current PR behaviour).
Hope that makes sense
Happy to discuss on Slack when next free

Reply:
My initial point was, that as enduser I don't understand what's possible or not. This feature should `override global prefix for a Tenant`, but if not defined use default.

I disagree with the annotation, however me and dario are just different in these approaches. Please make a very clear description of what the field does and it's implications for endusers. Leave it as CRD spec.

Reply:
I have pushed change for `overriding global prefix for a Tenant` , if omitted, will use global setting.